spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -103,7 +111,7 @@ private Metric createMetricToTransmit(TemporaryMetricsAggregation tma) {
         metricToTransmit.setRequestMethod(tma.getRequestMethod());
         metricToTransmit.setType(tma.getMetricType());
         metricToTransmit.setHttpResponseCode(tma.getHttpResponseCode());
-        metricToTransmit.setDateOfApiCall(Date.valueOf(LocalDate.now()));
+        metricToTransmit.setDateOfApiCall(java.util.Date.from(cetTimeNow.toInstant()));
```

Comment:
The previous code was setting just the date, meaning that the time was always set as `00:00:00`. The new code sets the current time instead. However, as on admin side just the date is stored in the DB, there seems to be now improvement when sending the current time.

Reply:
The problem here is that this could lead to bugs. With the old approach we are generating two time at different moments in the logic. It might be unlikely, but if one time is taken just before midnight and the other time just after midnight, we would have a time inconsistency problem which generates wrong metrics.

In my new approach, I take a time only once and derive any subsequent time information from it.